src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -585,9 +598,19 @@ func (c *controller) UploadLogs(ctx context.Context, cancellog context.CancelFun
 	for {
 		select {
 		case <-ctx.Done():
+			var retry = 0
+			var err error
 			if podName != "" {
 				c.log.Infof("Upload final controller and cluster logs before exit")
-				_ = c.uploadSummaryLogs(podName, c.Namespace, controllerLogsSecondsAgo, status.HasError())
+				for {
+					err = c.uploadSummaryLogs(podName, c.Namespace, controllerLogsSecondsAgo, status.HasError())
+					if err == nil || retry >= NumRetrySendingLogs {
```

Comment:
no need, retry till it is running, why matters ?

Reply:
why not to use WaitForPredicate ? and just create local function as 
func() bool {
   err = c.uploadSummaryLogs(podName, c.Namespace, controllerLogsSecondsAgo, status.HasError())
   if err != nil {
   log.smth
}
return err == nil
}

Reply:
10 seconds is too little. Even when it fails, the command takes a long time. I choose 5 minutes because it's the existing interval for collecting controller logs and it fits well with the times I see for the cluster to get ready

Reply:
I will use the predicate but as for the infinite retry I don't think it is good idea. we do want the controller to die at some point and not block forever. There is no guarantee that the oc makes it at all in some edge cases